packages/lisk-p2p/src/peer.ts

```diff
@@ -134,73 +259,40 @@ export class Peer {
 		});
 	}
 
-	public get peerConfig(): PeerConfig {
-		return this._peerConfig;
-	}
-
-	public get id(): string {
-		return this._id;
-	}
-
-	/**
-	 * This is not a declared as a setter because this method will need
-	 * invoke an async RPC on the socket to pass it the new node status.
-	 */
-	public applyNodeStatus(value: P2PNodeStatus | undefined): void {
-		this._nodeStatus = value;
-	}
-
-	public get nodeStatus(): P2PNodeStatus | undefined {
-		return this._nodeStatus;
-	}
-
-	public set inboundSocket(value: any) {
-		this._inboundSocket = value;
-	}
-
-	public get inboundSocket(): any {
-		return this._inboundSocket;
-	}
-
-	public set outboundSocket(value: any) {
-		this._outboundSocket = value;
+	private _createOutboundSocket(): void {
+		if (!this._outboundSocket) {
```

Comment:
prefer early return.
```
if (this._outboundSocket) {
  return;
}
```
Also, should this be fail safe to be called twice, or just throw an error instead of making it safe

Reply:
I turned this into a pure function so that should solve this issue.